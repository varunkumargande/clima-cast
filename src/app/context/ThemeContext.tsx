// src/context/ThemeContext.tsx
"use client";
import React, { createContext, useContext, ReactNode, useState } from "react";
import { Theme, ThemeContextProps, ThemeProviderProps } from "../types/types";

const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);

export const ThemeProvider: React.FC<ThemeProviderProps<ReactNode>> = ({
  children,
}) => {
  const [theme, setTheme] = useState<Theme>("light");

  const toggleTheme = () => {
    setTheme((prevTheme) => (prevTheme === "light" ? "dark" : "light"));
    document.documentElement.classList.toggle("dark", theme !== "dark");
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const context = useContext(ThemeContext);

  if (!context) {
    throw new Error("useTheme must be used within a ThemeProvider");
  }

  return context;
};
