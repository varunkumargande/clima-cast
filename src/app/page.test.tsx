import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Page from "@/app/page";
import { ThemeProvider } from "./context/ThemeContext";

describe("Page", () => {
  it("renders a heading", () => {
    render(
      <ThemeProvider>
        <Page />
      </ThemeProvider>
    );

    const heading = screen.getByRole("heading", { level: 1 });

    expect(heading).toBeInTheDocument();
  });
});

describe("Page Component", () => {
  it("matches snapshot", () => {
    // Render the Page component
    const tree = renderer
      .create(
        <ThemeProvider>
          <Page />
        </ThemeProvider>
      )
      .toJSON();

    // Compare the rendered component with a snapshot
    expect(tree).toMatchSnapshot();
  });
});
