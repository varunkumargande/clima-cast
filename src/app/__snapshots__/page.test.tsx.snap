// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Component matches snapshot 1`] = `
<main
  className="flex flex-col items-center justify-between mt-[139px] pb-3 mb-5"
>
  <img
    alt="Clima Cast"
    data-nimg="1"
    decoding="async"
    height={100}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Flogo.png&w=256&q=75"
    srcSet="/_next/image?url=%2Flogo.png&w=128&q=75 1x, /_next/image?url=%2Flogo.png&w=256&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={100}
  />
  <h1
    className="text-3xl font-bold mb-4"
  >
    Clima Cast
  </h1>
  <div />
</main>
`;
